texar/modules/classifiers/conv_classifiers.py

```diff
@@ -241,3 +241,17 @@ def layer_names(self) -> List[str]:
         r"""A list of uniquified layer names.
         """
         return self._encoder.layer_names
+
+    @property
+    def output_size(self) -> int:
+        r"""Return the final dimension of the output :attr:`logit`,
+        if the final dimension cannot be computed, return 1.
+        """
+        if self._hparams.num_classes > 1:
+            logit_dim = self._hparams.num_classes
+        elif self._hparams.num_classes == 1:
+            logit_dim = 1
+        else:
+            raise NameError("logit_dim cannot be defined "
+                            "if self._hparams.num_classes < 1")
```

Comment:
1. The error msg should not refer to `logit_dim` which is a local variable and should not be exposed to users.

2. I don't think a `NameError` is appropriate here. Probably `AttributeError`